File: 프로그래머스/2/42584. 주식가격/주식가격.js
function solution(prices) {
    let stack = [];
    let result = new Array(prices.length).fill(0);
    for (let i = 0; i < prices.length; i++) {
        while (stack.length > 0 && prices[stack.at(-1)] > prices[i]) {
            const top = stack.pop();
            result[top] = i - top;
        }
        stack.push(i);
    }
    
    while (stack.length > 0) {
        const top = stack.pop();
        result[top] = prices.length - 1 - top;
    }
    return result;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    stack = []
    answer = [0] * len(prices)
    for i in range(len(prices)):
        while stack and prices[stack[-1]] > prices[i]:
            j = stack.pop()
            answer[j] = i-j
        stack.append(i)
    
    while stack:
        j = stack.pop()
        answer[j] = len(prices) - j - 1
        
    return answer